File: 2024/day01/main.js
const { readFileSync } = require('fs');

const left = [];
const right = [];

readFileSync('input', 'utf-8')
  .trim()
  .split('\n')
  .forEach((line) => {
    const f = line.split('   ').map(Number);
    left.push(f[0]);
    right.push(f[1]);
  });

// Part 1: 2164381
left.sort();
right.sort();
let sum = 0;
for (let i = 0; i < left.length; i++) {
  sum += Math.abs(left[i] - right[i]);
}
console.log(sum);

// Part 2: 20719933
sum = 0;
for (const x of left) {
  sum += x * right.filter((y) => y === x).length;
}
console.log(sum);

File: 2024/day01/main.py
left = []
right = []

for line in open('input').read().splitlines():
	f = line.split('   ')
	left.append(int(f[0]))
	right.append(int(f[1]))

# Part 1: 2164381
left.sort()
right.sort()
print(sum(abs(left[i] - right[i]) for i in range(len(left))))


# Part 2: 20719933
print(sum(x * right.count(x) for x in left))
